### PR TITLE
Catch exception when getting program details

### PR DESCRIPTION
**Pull request by mjfrey**
Add exception handling when getting program details for a conditional offer.

LEARNER-1944

### CODE REVIEW

**Comment by mjfrey** [comment]
@afzaledx please review

**Comment by mjfrey** [comment]
@staubina take a look.

**Review by afzaledx — APPROVED** [review]
LGTM

File: ecommerce/programs/conditions.py [hunk 1]
```diff
@@ -55,7 +55,11 @@ def is_satisfied(self, offer, basket):  # pylint: disable=unused-argument
 
         basket_skus = set([line.stockrecord.partner_sku for line in basket.all_lines()])
 
-        program = self.get_program(basket.site.siteconfiguration)
+        try:
+            program = self.get_program(basket.site.siteconfiguration)
+        except:  # pylint: disable=bare-except
```
Comment (afzaledx) [inline ecommerce/programs/conditions.py]: why not just the HttpNotFoundError ?

Comment (mjfrey) [reply ecommerce/programs/conditions.py]: That throws more than that error and we should catch them all.

**Review by staubina — APPROVED** [review]
👍

**Pull request merged**